src/storage-types/src/controller.rs

```diff
@@ -161,7 +162,7 @@ impl RustType<ProtoDurableCollectionMetadata> for DurableCollectionMetadata {
 }
 
 #[derive(Debug)]
-pub enum StorageError {
+pub enum StorageError<T> {
```

Comment:
What happened to `enum StorageError<T = Timestamp>`? Seems like that'd reduce a lot of this PR.

Reply:
It turned out not to have a big impact at all. If some function/trait/struct already had some generic timestamp parameter, then it needed to pass that same generic parameter to `StorageError` so that the types would match up. That ended up being every single usage of `StorageError` in the storage crates. There were a couple of usages in `mz_adapter` where `enum StorageError<T = Timestamp>` would have saved some types, but the total saved lines were 4.

This should show the actual difference it would have made: https://github.com/MaterializeInc/materialize/compare/3cb3ec799fe85142a6b54588b540ee76906e2583..8fb3e0478d103f9c880f77ff32593d0e912127af. At that point I thought I might as well explicitly pass in the type everywhere.